src/remix/remix-wizard.ts

```diff
@@ -0,0 +1,122 @@
+import clack from '@clack/prompts';
+import {
+  addSentryCliRc,
+  askForProjectSelection,
+  askForSelfHosted,
+  askForWizardLogin,
+  confirmContinueEvenThoughNoGitRepo,
+  ensurePackageIsInstalled,
+  getPackageDotJson,
+  installPackage,
+  isUsingTypeScript,
+  printWelcome,
+} from '../utils/clack-utils';
+import { hasPackageInstalled } from '../utils/package-json';
+import { WizardOptions } from '../utils/types';
+import {
+  initializeSentryOnEntryClient,
+  initializeSentryOnEntryServer,
+  updateBuildScript,
+  instrumentRootRoute,
+  isRemixV2,
+  loadRemixConfig,
+} from './sdk-setup';
+import { debug } from '../utils/debug';
+import { traceStep, withTelemetry } from '../telemetry';
+
+export async function runRemixWizard(options: WizardOptions): Promise<void> {
+  return withTelemetry(
+    {
+      enabled: options.telemetryEnabled,
+      integration: 'remix',
+    },
+    () => runRemixWizardWithTelemetry(options),
+  );
+}
+
+async function runRemixWizardWithTelemetry(
+  options: WizardOptions,
+): Promise<void> {
+  printWelcome({
+    wizardName: 'Sentry Remix Wizard',
+    promoCode: options.promoCode,
+  });
+
+  await confirmContinueEvenThoughNoGitRepo();
+
+  const remixConfig = await loadRemixConfig();
+  const packageJson = await getPackageDotJson();
+
+  // We expect `@remix-run/dev` to be installed for every Remix project
+  await ensurePackageIsInstalled(packageJson, '@remix-run/dev', 'Remix');
+
+  const { url: sentryUrl } = await askForSelfHosted(options.url);
+
+  const { projects, apiKeys } = await askForWizardLogin({
+    promoCode: options.promoCode,
+    url: sentryUrl,
+    platform: 'javascript-remix',
+  });
+
+  const selectedProject = await askForProjectSelection(projects);
+
+  await traceStep('Install Sentry SDK', () =>
+    installPackage({
+      packageName: '@sentry/remix',
+      alreadyInstalled: hasPackageInstalled('@sentry/remix', packageJson),
+    }),
+  );
+
+  const dsn = selectedProject.keys[0].dsn.public;
+
+  const isTS = isUsingTypeScript();
+  const isV2 = isRemixV2(remixConfig, packageJson);
+
+  await addSentryCliRc(
+    apiKeys.token,
+    selectedProject.organization.slug,
+    selectedProject.name,
+  );
+
+  await traceStep('Update build script for sourcemap uploads', async () => {
+    try {
+      await updateBuildScript();
+    } catch (e) {
+      clack.log
+        .warn(`Could not update build script to generate and upload sourcemaps.
+  Please update your build script manually using instructions from https://docs.sentry.io/platforms/javascript/guides/remix/sourcemaps/`);
+      debug(e);
+    }
+  });
+
+  await traceStep('Instrument root route', async () => {
+    try {
+      await instrumentRootRoute(isV2, isTS);
+    } catch (e) {
+      clack.log.warn(`Could not instrument root route.
+  Please do it manually using instructions from https://docs.sentry.io/platforms/javascript/guides/remix/`);
```

Comment:
I think for now this is fine. Once we put the wizard instructions into docs, I think it's likely that we'll refactor the remix docs so that the manual setup will be in its own page (similarly to Next and Svelte). We just have to adjust the links then. I added a task in the follow-up issue (https://github.com/getsentry/sentry/issues/54615)